### ISSUE TITLE
Cannot apply extension method in given instance for Array

### ISSUE THREAD

**Issue by HuangBoHong** [lampepfl/dotty#12790]
## Compiler version

3.0.1-RC1

## Minimized code

```Scala
val arr = Array(1, 2, 3)

trait SeqOps {
  extension[T](seq: collection.Seq[T]) {
    def randomOption: Option[T] = seq.size match {
      case 0 => None
      case size => Some(seq(scala.util.Random.nextInt(size)))
    }
  }
}
given seqOps: SeqOps with {}

extension[T](seq: collection.Seq[T]) {
  def random: T = seq.randomOption.get
}

arr.random // OK
arr.randomOption // Error: value randomOption is not a member of Array[Int]
(arr: collection.Seq[Int]).randomOption // OK
```

**Comment by smarter** [comment]
An Array is not a Seq, `(arr: collection.Seq[Int]).randomOption` works because it finds the implicit conversion `scala.Predef.wrapIntArray`, but `arr.randomOption` won't work, this is as expected.

**Comment by smarter** [comment]
Oh, and `arr.random` works because `random` is in scope at that point, so the compiler tries it directly even though the receiver type doesn't match `arr`, but I wouldn't rely on that.

**Comment by som-snytt** [comment]
I don't understand the explanation. The docs says it a simple rewrite. `m(arr)`, so what does it mean to "rely" on it? Also, `o.m(arr)` if `o` is visible, which it is in the example, so why should it fail?

**Comment by smarter** [comment]
I mean, I wouldn't rely on combining extension methods with implicit conversions, even if it happens to work in some example because of how your scopes are set up. In `arr.randomOption` there's no randomOption in scope, so we do an implicit search instead, but that search is constrained to looking for a conversion from Array to something that has a randomOption method, so it won't find the extension in seqOps.

**Comment by som-snytt** [comment]
That is not what the docs say. "The doc says" (sorry) that should try `myGivenInLexicalScope.m(arr)` as a second step before consulting implicit scope. Am I misreading? I haven't experimented yet.

**Comment by smarter** [comment]
Ah indeed, the doc says:
> 2. The extension method is a member of some given instance that is visible at the point of the reference.

But does not mention whether the extension method has to be applicable or not, so either the doc or the implementation should be updated /cc @odersky .

**Comment by smarter** [comment]
My preference would be to disallow `arr.random` from compiling even though `random` and an implicit conversion to its receiver type are in scope, because this is basically equivalent to chaining implicit conversions which we don't normally do since it's both expensive and confusing.

**Comment by odersky** [comment]
I agree that combining extensions with conversions on the same value should be disallowed.
